src/engine/Timer.ts

```diff
@@ -37,6 +47,32 @@ export class Timer {
     this.interval = interval || this.interval;
     this.fcn = fcn || this.fcn;
     this.repeats = repeats || this.repeats;
+
+    this._callbacks = {};
+
+    if (fcn) {
+      this.on(fcn);
+    }
+  }
+
+  /**
+   * Adds a new callback to be fired after the interval is complete
+   * @param fcn The callback to be added to the callback list, to be fired after the interval is complete.
+   */
+  public on(fcn: () => void) {
+    const key = fcn.toString();
```

Comment:
Please use an array of functions, and not a dictionary. `this._callbacks.push(fcn)`. Functions have a unique reference and can be compared that when when `off(...)` is called. 

```typescript
private _callbacks: (() => void)[];
...
public off(fcn: () => void) {
  const index = this._callbacks.indexOf(fcn);
  if (index > -1) {
    this._callbacks.splice(index, 1);
  }
}	  
```

I'm slightly concerned about using the function code as the key, it would mean that functions that are syntactically the same but not, contextually the same could overwrite each other. This is probably unlikely but it would be a tricky bug. Example of the bug:

https://codepen.io/eonarheim/pen/MWWWvGb?editors=0011

Reply:
I was thinking of that also, nice tip about the reference! Thank you!